driver/ppm_fillers.c

```diff
@@ -7978,3 +7978,81 @@ int f_sched_prog_fork(struct event_filler_arguments *args)
 	return add_sentinel(args);
 }
 #endif
+
+int f_sys_prctl_x(struct event_filler_arguments *args)
+{
+	int res;
+	int retval;
+	char *name = NULL;
+	syscall_arg_t option;
+	syscall_arg_t arg2;
+
+	/* Parameter 1: res (type: PT_ERRNO) */
+	retval = (int64_t)syscall_get_return_value(current, args->regs);
+	res = val_to_ring(args, retval, 0, false, 0);
+	CHECK_RES(res);
+
+	/*
+	 * option
+	 */
+	syscall_get_arguments_deprecated(current, args->regs, 0, 1, &option);
+	option = prctl_options_to_scap(option);
+	res = val_to_ring(args, option, 0, false, 0);
+	CHECK_RES(res);
+
+	/*
+	 * arg2
+	 */
+	syscall_get_arguments_deprecated(current, args->regs, 1, 1, &arg2);
+
+	switch(option){
+		case PPM_PR_GET_NAME:
+		case PPM_PR_SET_NAME:
+			/*
+			 * arg2_str
+			 */
+			if(likely(ppm_strncpy_from_user(args->str_storage, (const void __user *)arg2, PPM_MAX_PATH_SIZE) >= 0))
+			{
+				name = args->str_storage;
+				name[PPM_MAX_PATH_SIZE - 1] = '\0';
+			}
+			res = val_to_ring(args, (int64_t)(long)name, 0, false, 0);
+			CHECK_RES(res);
+			/*
+			 * arg2_int
+			 */
+			res = val_to_ring(args, 0, 0, false, 0);
+			CHECK_RES(res);
+			break;
+		case PPM_PR_GET_CHILD_SUBREAPER:
+			{
+				int reaper_attr = 0;
+				/* Parameter 3: arg2_str (type: PT_CHARBUF) */
+				res = push_empty_param(args);
+				CHECK_RES(res);
+				/* Parameter 4: arg2_int (type: PT_INT64) */
+				if(unlikely(ppm_copy_from_user(&reaper_attr, (void *)arg2, sizeof(reaper_attr))))
+				{
+					reaper_attr = 0;
+				}
+				res = val_to_ring(args, (s64)reaper_attr, 0, false, 0);
+				CHECK_RES(res);
+			}
+			break;
+		case PPM_PR_SET_CHILD_SUBREAPER:
```

Comment:
Leftover case?

Reply:
In reality I kept this because the `PPM_PR_SET_CHILD_SUBREAPER` is basically the default case. If you want I can remove it. 😄

Reply:
Oh it's ok! I'll put it in all 3 drivers then!